com.arm.cmsis.pack.data/src/com/arm/cmsis/pack/data/CpExample.java

```diff
@@ -64,13 +64,22 @@ public String getBoardId() {
 
 	@Override
 	public String getLoadPath(String environmentName) {
-		if(environmentName == null)
+		return getEnvironmentAttribute(environmentName, CmsisConstants.LOAD);
+	}
+
+	@Override
+	public String getProjectFolder(String environmentName) {
+		return getEnvironmentAttribute(environmentName, CmsisConstants.FOLDER);
+	}
+
+	private String getEnvironmentAttribute(String environmentName, String attributeName) {

```

Comment:
please change visibility to protected and add a comment. CpItem-based classes should be kept open for extension.